File: src/ScheduleTime.Domain/Factories/BarbershopFactory.cs
using ScheduleTime.Domain.Entities;
using ScheduleTime.Domain.Factories.Interfaces;

namespace ScheduleTime.Domain.Factories
{
    public class BarbershopFactory : IHairServiceFactory
    {
        public HairService CreateServiceHair(Customer customer, DateTime date)
        {
            return new BarberShop(customer, date);
        }
    }
}

File: java/schedule-time-event/src/main/java/com/gearguru/schedule_time_event/Domain/Factories/BarberShopFactory.java
package com.gearguru.schedule_time_event.Domain.Factories;

import com.gearguru.schedule_time_event.Domain.Entities.Barbershop;
import com.gearguru.schedule_time_event.Domain.Entities.Schedule;
import com.gearguru.schedule_time_event.Domain.Interfaces.IScheduleFactory;

public class BarberShopFactory implements IScheduleFactory {
    @Override
    public Schedule scheduleTime() {
        return new Barbershop();
    }
}
